### PR TITLE
rename structural records: tuple -> struct

### PR DESCRIPTION
**Pull request by nickchapman-da**
Update daml-lf spec to align with name change in proto/haskell/scala files.
https://github.com/digital-asset/daml/pull/3658

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags, if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: struct-files.list [hunk 1]
```diff
@@ -0,0 +1,58 @@
+compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Base.hs
```
Comment (remyhaemmerle-da) [inline struct-files.list]: What is the purpose of this file ?

Comment (nickchapman-da) [reply struct-files.list]: It's the haskell definition of the DAML-LF AST

File: daml-lf/spec/daml-lf-1.rst [hunk 2]
```diff
@@ -475,7 +475,7 @@ DAML-LF as a "not an Ident" notation, so should not be considered for
 future addition to allowed identifier characters.
 
 In the following, we will use identifiers to represent *built-in
-functions*, term and type *variable names*, record and tuple *field
+functions*, term and type *variable names*, record and struct *field
```
Comment (hurryabit) [inline daml-lf/spec/daml-lf-1.rst]: We should explain somewhere that "record" means nominal record and "struct" means structural records.

**Comment by nickchapman-da** [comment]
> spec looks good. Thanks.
> 
> Still missing changes in
> 
> * com/digitalasset/navigator/console/Pretty.scala
> * com/digitalasset/navigator/json/ApiCodecVerbose.scala
> * com/digitalasset/navigator/model/converter/LedgerApiV1.scala
> * com/digitalasset/daml/lf/validation/TypingSpec.scala
> * daml-lf/encoder/src/test/scala/com/digitalasset/daml/lf/testing/archive/EncodeV1Spec.scala

done

File: daml-lf/spec/daml-lf-1.rst (hunk 2 above)
Comment (nickchapman-da) [reply daml-lf/spec/daml-lf-1.rst]: done

**Comment by nickchapman-da** [comment]
> Please the following line in the history part of the spec (around line 270 in daml-lf/spec/daml-lf-1.rst):
> 
> * **Rename** `Tuple` to `Struct`

done

**Pull request merged**